src/quantuminspire/projectq/backend_qx.py

```diff
@@ -63,8 +64,12 @@ def __init__(self, num_runs: int = 1024, verbose: int = 0, quantum_inspire_api:
         self._measured_ids: List[int] = []
         self._allocated_qubits: Set[int] = set()
         self._max_qubit_id: int = -1
-        if not quantum_inspire_api:
-            raise RuntimeError("Api is required")
+        if quantum_inspire_api is None:
+            try:
+                quantum_inspire_api = QuantumInspireAPI()
+            except ApiError as ex:
```

Comment:
According to the docstring on `QuantumInspireAPI`, `ApiError` is raised either when authentication is not available or when the schema could not be loaded. The reraise below suggests to the user that the error is always that there are no token credentials saved. That seems inconsistent to me. This means that if the reason for the `ApiError` is not that there is no authentication, the user will think they need to specify authentication while they already have, greatly confusing the user.

Perhaps this means the exceptions in the `QuantumInspireAPI` shoudl not be of the same class, and a new class should be introduced, `AuthenticationError` or something like that?

Reply:
Agree, I'll add AuthenticationError for this.